docs/dg/dev/integrate-and-configure/integrate-svg-sprite-asset.md

```diff
@@ -0,0 +1,220 @@
+        const match = twigContent.match(spacelessRegex);
+
+        if (!match || !match[1]) {
+            throw new Error(`Could not find content within spaceless block in the Twig file: ${usedPath}`);
+        }
+
+        const svgContent = `<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" style="display: none;">
+${match[1]}
+</svg>`;
+
+        const targetDir = path.dirname(targetPath);
+        await fs.mkdir(targetDir, { recursive: true });
+
+        await fs.writeFile(targetPath, svgContent, 'utf8');
+
+        console.info('Icon sprites successfully extracted to', targetPath);
+    } catch (error) {
+        console.error('Error extracting icon sprites:', error.message);
+    }
+};
+
+module.exports = extractIconSprites;
+```
+
+---
+
+5. Update Frontend Settings
+
+**File:**  
+`frontend/settings.js`
+
+**Add the following inside your `globalSettings` paths:**
+```js
+    iconSprite: {
+        sources: [
+            './src/Pyz/Yves/ShopUi/Theme/default/components/atoms/icon-sprite/icon-sprite.twig',
+            './vendor/spryker-shop/shop-ui/src/SprykerShop/Yves/ShopUi/Theme/default/components/atoms/icon-sprite/icon-sprite.twig',
+        ],
+        target: './frontend/assets/global/default/icons/sprite.svg',
+    },
+```
+**And inside `getAppSettingsByTheme`:**
+```js
+    iconSprite: globalSettings.paths.iconSprite,
+```
+
+---
+
+6. Update the Frontend Build Config
+
+**File:**  
+`frontend/configs/development.js`
+
+**Add or update:**
+```js
+let isExtractIconSpritesEnabled = false;
+let extractIconSprites = null;
+
+try {
+    extractIconSprites = require('../libs/icon-sprite-extractor');
+    isExtractIconSpritesEnabled = true;
+} catch (e) {
+    console.info('Icon sprite extraction is disabled.');
+}
+
+const getConfiguration = async (appSettings) => {
+    // ...existing code...
+
+    if (isExtractIconSpritesEnabled) {
+        try {
+            await extractIconSprites({
+                sourcePath: appSettings.paths.iconSprite.sources.map((src) => join(process.cwd(), src)),
+                targetPath: join(process.cwd(), appSettings.paths.iconSprite.target),
+            });
+        } catch (error) {
+            console.error('Error extracting icon sprites:', error);
+        }
+    }
+
+    // ...rest of your function...
+};
+```
+
+---
+
+7. Update `.gitignore`
+
+Add this line to avoid committing the generated sprite:
+
+```
+/frontend/assets/global/default/icons/sprite.svg
+```
+
+---
+
+8. Rebuild the Frontend
+
+Run:
+```bash
+console frontend:project:install-dependencies
+console frontend:yves:build -e production
+```
+
+9. Rebuild twig cache
+```bash
+console t:c:w
+```
+
+---
+
+**You’re all set!**  
+The system now automatically extracts SVG symbols from your Twig file and generates a sprite for use in the frontend. Icons can be managed simply by editing `icon-sprite.twig` and running the build.
```

Comment:
My suggestion is to avoid such a long and complicated way
we need to suggest 
1. create a file (or better of course have it in core)
2. copy-paste svg from HTML to this file
3. work with this file
and that's it